spring-kafka/src/main/java/org/springframework/kafka/support/serializer/ErrorHandlingDeserializer2.java

```diff
@@ -92,10 +101,19 @@ public ErrorHandlingDeserializer2(Deserializer<T> delegate) {
 		this.delegate = setupDelegate(delegate);
 	}
 
+	/**
+	 * @since 1.3
+	 * @deprecated in favor of {@link ErrorHandlingDeserializer2#setFailedDeserializationFunction(Function)}.
+	 */
+	@Deprecated
 	public void setFailedDeserializationFunction(BiFunction<byte[], Headers, T> failedDeserializationFunction) {
 		this.failedDeserializationFunction = failedDeserializationFunction;
```

Comment:
Well, this one can be as a delegate from inline `Function<FailedDeserialization, T>` implementation and therefore can be deprecated:
```
this.failedDeserializationMappingFunction =
          return (failed) -> failedDeserializationFunction.apply(failed.getData(), failed.getHeaders());
```